<a href="https://colab.research.google.com/github/Ksenia-90/probability_theory/blob/lesson_8/lesson_8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Урок 8. Дисперсионный анализ. Логистическая регрессия**

In [52]:
import numpy as np
import pandas as pd
from scipy import stats
%matplotlib inline

**Задача 1**

Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов:

* football_players = [173, 175, 180, 178, 177, 185, 183, 182]
* hockey_players = [177, 179, 180, 188, 177, 172, 171, 184, 180]
* lifters = [172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]


In [3]:
football_players = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hockey_players = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180]) 
lifters = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [54]:
football_players_mean = football_players.mean()
hockey_players_mean = hockey_players.mean()
lifters_mean = lifters.mean()
print(f'Cреднее значение:') 
print(f'football_players = {fp_mean}')
print(f'hockey_players =  {np.round(hp_mean, 3)}')
print(f'lifters =  {np.round(lf_mean, 3)}')

Cреднее значение:
football_players = 179.125
hockey_players =  178.667
lifters =  172.727


In [33]:
n_1 = football_players.shape[0]
n_2 = hockey_players.shape[0] 
n_3 = lifters.shape[0]
n_1, n_2, n_3

(8, 9, 11)

In [55]:
height = np.concatenate([football_players, hockey_players, lifters])
height_mean = height.mean()
print(f'Cреднее значение общее {np.round(height_mean, 3)}')

Cреднее значение общее 176.464


Найдем значение S2_b и S2_w

In [34]:
S2_b = n_1 * (football_players_mean - height_mean) ** 2 + n_2 * (hockey_players_mean - height_mean) ** 2 + n_3 * (lifters_mean - height_mean) ** 2
print(f'S2_b = {np.round(S2_b, 3)}')

S2_b = 253.907


In [35]:
#внутригрупповоеотклонение
S2_w = ((football_players - football_players_mean)**2).sum() + ((hockey_players - hockey_players_mean) ** 2).sum() + ((lifters - lifters_mean) ** 2).sum()
print(f'S2_w = {np.round(S2_w, 3)}')

S2_w = 577.057


In [36]:
S2_b + S2_w, ((height - height_mean) ** 2).sum()

(830.964285714286, 830.9642857142854)

In [45]:
k = 3
n = n_1 + n_2 + n_3
k_1 = k-1
k_2 = n - k
k_1, k_2, n

(2, 25, 28)

In [40]:
sigma2_b = S2_b / (k - 1)
sigma2_w = S2_w / (n - k)

sigma2_b, sigma2_w

(126.9537337662339, 23.08227272727273)

In [41]:
T = sigma2_b / sigma2_w
T

5.500053450812598

In [42]:
# возьмем уровень значимости 5% (или 0,05)
alpha = 0.05
F_crit = stats.f.ppf(1 - alpha, (k - 1), (n - k))
F_crit


3.3851899614491687

In [49]:
p = 1 - stats.f.cdf(T, k_1,k_2)
p

0.010482206918698611

T (5.5) > F_crit (3.4), можно сделать вывод, что отличие среднего роста среди взрослых футболистов, хоккеистов и штангистов является статистически значимым.

**Задача 2**

Прикрепите ссылку на курсовой проект на github (см. выбор темы в ДЗ к Уроку 6).